### PR TITLE
[MED-15550] Istio Sidecar configuration resource

### PR DESCRIPTION
**Pull request by matkam**
Creates a Sidecar resource for deployment targets configured with `istioSidecar`.

Tested using a target with:
```
istioSidecar:
  egressHosts:
  - istio-system/*
```

Picchu created a Sidecar:
```
apiVersion: networking.istio.io/v1beta1
kind: Sidecar
metadata:
  labels:
    app: echo
    app.kubernetes.io/name: echo
    picchu.medium.engineering/app: echo
    picchu.medium.engineering/ownerName: echo-production
    picchu.medium.engineering/ownerType: releasemanager
  name: echo
  namespace: echo-production
spec:
  egress:
  - hosts:
    - istio-system/*
```

Verified the Sidecar with:
```
> kctl sandbox-reef -n echo-production exec -it testing1-wv8w4 -c istio-proxy -- curl http://127.0.0.1:15000/config_dump | grep echo-sandbox
(no output)
```

And after adding
```
spec:
  egress:
  - hosts:
    - istio-system/*
    - echo-sandbox/*
```

The Sidecar began receiving configuration updates from `echo-sandbox`:
```
> kctl sandbox-reef -n echo-production exec -it testing1-wv8w4 -c istio-proxy -- curl http://127.0.0.1:15000/config_dump | grep echo-sandbox
      "name": "outbound|4242|main-20200811-172426-1f382178e8|echo.echo-sandbox.svc.cluster.local",
       "service_name": "outbound|4242|main-20200811-172426-1f382178e8|echo.echo-sandbox.svc.cluster.local"
         "config": "/apis/networking.istio.io/v1alpha3/namespaces/echo-sandbox/destination-rule/echo",
      "name": "outbound|4242||echo.echo-sandbox.svc.cluster.local",
...
```

### CODE REVIEW

File: pkg/apis/picchu/v1alpha1/revision_types.go [hunk 1]
```diff
@@ -24,6 +24,10 @@ type Istio struct {
 	TrafficPolicy *istio.TrafficPolicy `json:"trafficPolicy,omitempty"`
```
Comment (silverlyra) [inline pkg/apis/picchu/v1alpha1/revision_types.go:24]:
```suggestion
	TrafficPolicy *istio.TrafficPolicy `json:"trafficPolicy,omitempty"`
	Sidecar       *IstioSidecar        `json:"sidecar,omitempty"`
```

let’s keep the Istio configuration together

File: pkg/apis/picchu/v1alpha1/revision_types.go [hunk 2]
```diff
@@ -128,7 +132,8 @@ type RevisionTarget struct {
 	Ports               []PortInfo        `json:"ports,omitempty"`
 	Env                 []corev1.EnvVar   `json:"env,omitempty"`
 
-	Istio *Istio `json:"istio,omitempty"`
+	Istio        *Istio        `json:"istio,omitempty"`
+	IstioSidecar *IstioSidecar `json:"istioSidecar,omitempty"`
```
Comment (silverlyra) [inline pkg/apis/picchu/v1alpha1/revision_types.go:136]:
```suggestion
```

File: pkg/controller/releasemanager/plan/syncApp.go [hunk 3]
```diff
@@ -73,14 +74,19 @@ func (p *SyncApp) Apply(
...
+			return err
+		}
+	}
 
+	destinationRule := p.destinationRule()
 	if err := plan.CreateOrUpdate(ctx, log, cli, destinationRule); err != nil {
 		return err
 	}
 
+	prometheusRule := p.prometheusRule()
```
Comment (silverlyra) [inline pkg/controller/releasemanager/plan/syncApp.go:89]: 👍 ty for this cleanup – tidier

File: pkg/controller/releasemanager/plan/syncApp.go [hunk 4]
```diff
@@ -570,3 +577,24 @@ func (p *SyncApp) prometheusRule() *monitoringv1.PrometheusRule {
...
+	}
+
+	return &istioclient.Sidecar{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.App,
+			Namespace: p.Namespace,
+			Labels:    p.Labels,
+		},
+		Spec: istio.Sidecar{
+			Egress: []*istio.IstioEgressListener{
```
Comment (dokipen) [inline pkg/controller/releasemanager/plan/syncApp.go:593]: Does this send all egress through the egress gateways automatically!?

Comment (matkam) [reply pkg/controller/releasemanager/plan/syncApp.go:593]: I wish. This only limits the xDS updates to Istio sidecars inside the namespace with the Sidecar resource.

**Pull request merged**